app/models/katello/content_view.rb

```diff
@@ -316,6 +317,14 @@ def duplicate_puppet_modules
       counts.select { |_k, v| v > 1 }.keys
     end
 
+    def check_non_composite_components
+      unless composite?
+        unless components.blank?
```

Comment:
@ehelms, are you thinking a scenario like the one below?  (e.g. view begins with 1 component and then is updated to contain 3)

```
hammer> content-view info --id 13
ID:                     13
Name:                   composite
Label:                  composite
Composite:              true
Description:            
Content Host Count:     0
Organization:           Default Organization
Yum Repositories:       
 1) ID:    9
    Name:  el6 x86_64
    Label: el6_x86_64
 2) ID:    10
    Name:  el6 x86_64
    Label: el6_x86_64
Docker Repositories:    

OSTree Repositories:    

Puppet Modules:         

Lifecycle Environments: 

Versions:               

Components:             
 1) ID:   3
    Name: tools el6 1.0
Activation Keys:

hammer> content-view update --organization-id 1 --id 13 --component-ids 2,3,4
Content view updated
hammer> content-view info --id 13
ID:                     13
Name:                   composite
Label:                  composite
Composite:              true
Description:            
Content Host Count:     0
Organization:           Default Organization
Yum Repositories:       
 1) ID:    7
    Name:  zoo
    Label: zoo
 2) ID:    8
    Name:  zoo
    Label: zoo
 3) ID:    9
    Name:  el6 x86_64
    Label: el6_x86_64
 4) ID:    10
    Name:  el6 x86_64
    Label: el6_x86_64
 5) ID:    12
    Name:  el7 x86_64
    Label: el7_x86_64
 6) ID:    11
    Name:  el7 x86_64
    Label: el7_x86_64
Docker Repositories:    

OSTree Repositories:    

Puppet Modules:         

Lifecycle Environments: 

Versions:               

Components:             
 1) ID:   2
    Name: zoo 1.0
 2) ID:   3
    Name: tools el6 1.0
 3) ID:   4
    Name: tools el7 1.0
Activation Keys:

hammer> 
```